src/realm/array.hpp

```diff
@@ -1491,7 +1315,7 @@ bool Array::find_optimized(int64_t value, size_t start, size_t end, size_t basei
                 bool value_is_null = (v == null_value);
                 if (c(v, value, value_is_null, find_null)) {
                     util::Optional<int64_t> v2(value_is_null ? util::none : util::make_optional(v));
-                    if (!find_action<action, Callback>(start2 + baseindex, v2, state, callback))
+                    if (!find_action<Callback>(start2 + baseindex, v2, state, callback))
```

Comment:
Minor thing: We can remove the explicit `Callback` template argument here, now that `action` is gone.